addons/website_blog/static/src/snippets/s_blog_posts/options.js

```diff
@@ -1,32 +1,115 @@
-odoo.define('website_blog.s_latest_posts_editor', function (require) {
+odoo.define('website_blog.s_blog_posts_options', function (require) {
 'use strict';
 
-var sOptions = require('web_editor.snippets.options');
+const options = require('web_editor.snippets.options');
+const dynamicSnippetOptions = require('website.s_dynamic_snippet_options');
+
 var wUtils = require('website.utils');
 
-sOptions.registry.js_get_posts_selectBlog = sOptions.Class.extend({
+const dynamicSnippetBlogPostsOptions = dynamicSnippetOptions.extend({
+    /**
+     *
+     * @override
+     */
+    init: function () {
+        this._super.apply(this, arguments);
+        this.modelName = 'blog.post';
+        this.templateName = 'blog_posts';
+    },
 
     //--------------------------------------------------------------------------
     // Private
     //--------------------------------------------------------------------------
 
     /**
+     *
      * @override
+     * @private
      */
-    _renderCustomXML: function (uiFragment) {
+    _computeWidgetVisibility: function (widgetName, params) {
+        if (widgetName === 'hover_effect_opt') {
+            return this.$target.get(0).dataset.templateKey === 'website_blog.dynamic_filter_template_blog_posts_big_picture';
+        }
+        return this._super.apply(this, arguments);
+    },
+    /**
+     * Fetches blogs.
+     * @private
+     * @returns {Promise}
+     */
+    _fetchBlogs: function () {
         return this._rpc({
             model: 'blog.blog',
             method: 'search_read',
-            args: [wUtils.websiteDomain(this), ['name']],
-        }).then(blogs => {
-            const menuEl = uiFragment.querySelector('[name="blog_selection"]');
-            for (const blog of blogs) {
-                const el = document.createElement('we-button');
-                el.dataset.selectDataAttribute = blog.id;
-                el.textContent = blog.name;
-                menuEl.appendChild(el);
+            kwargs: {
+                domain: wUtils.websiteDomain(this),
+                fields: ['id', 'name'],
             }
         });
     },
+    /**
+     *
+     * @override
+     * @private
+     */
+    _renderCustomXML: async function (uiFragment) {
+        await this._super.apply(this, arguments);
+        await this._renderBlogSelector(uiFragment);
+    },
+    /**
+     * Automatically assign class on template selection
+     * @override
+     */
+    _renderSelectUserValueWidgetButtons: async function (selectUserValueWidgetElement, data) {
+        for (let id in data) {
+            const button = document.createElement('we-button');
+            button.dataset.selectDataAttribute = id;
+            if (id.startsWith("website_blog.dynamic_filter_template_")) {
+                button.dataset.selectClass = id.replace("website_blog.dynamic_filter_template_", "s_");
```

Comment:
:eyes: ?

Reply:
The CSS currently relies on a `class` related to the selected element template to be applied on the `section` itself.
This is now done in `selectDataAttribute` because using `selectClass` prevented other behaviors from remaining correct.